### PR TITLE
delete-image and terminate added

### PR DESCRIPTION
**Pull request by twarnock2w**
Also remove check in get_project_id that short circuits the lookup if the names match but the project_id isn't populated.

### CODE REVIEW

File: cloudendure/cloudendure.py [hunk 1]
```diff
@@ -93,8 +93,6 @@ def get_project_id(self, project_name: str = "") -> str:
             return ""
 
         print(f"project name self: {self.project_name} - project name: {project_name}")
-        if self.project_name and self.project_name == project_name:
```
Comment (mbeacom) [inline cloudendure/cloudendure.py]: Why are we removing this? If the active project name matches the tracked project name and project_id is already available, why do we want to fetch it again? 🤔

File: cloudendure/cloudendure.py [hunk 2]
```diff
@@ -774,6 +773,83 @@ def gen_terraform(
...
+                    "machineCloudId", "NONE"
+                )
+                if _machine == source_props.get(
+                    "name", "NONE"
+                ) or _machine == source_props.get("machineCloudId", "NONE"):
+                    replica: str = machine.get("replica")
+                    if replica:
+                        f"{ref_name} has a launched machine: {replica}.  Terminating."
+                        data_dict: Dict[str, Any] = {}
+                        data_dict["replicaIDs"] = [replica]
```
Comment (mbeacom) [inline cloudendure/cloudendure.py]: We can just default to `data_dict: Dict[str, Any] = {"replicaIDs": [replica]}` here

File: cloudendure/cloudendure.py [hunk 3]
```diff
@@ -774,6 +773,80 @@ def gen_terraform(
...
+                        if delete_response.status_code != 202:
+                            print(
+                                f"Response code: {delete_response.status_code}\n{ref_name} replica {replica} did not terminate.\n{delete_response.text}"
+                            )
+                        else:
+                            print(
+                                f"Terminated {ref_name}\n{json.loads(delete_response.text)}"
+                            )
+                    else:
+                        f"{ref_name} does not have a launched machine"
```
Comment (mbeacom) [inline cloudendure/cloudendure.py]: missing `print()`

File: cloudendure/cloudendure.py [hunk 4]
```diff
@@ -774,6 +773,80 @@ def gen_terraform(
...
+
+        return True
```
Comment (mbeacom) [inline cloudendure/cloudendure.py]: Should we return `False` if the termination isn't successful?

File: cloudendure/cloudendure.py (hunk 1 above)
Comment (twarnock2w) [reply cloudendure/cloudendure.py]: It is returning self.project_id before setting it in the use case of just specifying the project name in the environment.  This leads to an empty project ID which causes some weird behavior:

project name self: tw-test - project name: tw-test
Checking Project - Name: (tw-test) - Dry Run: (False)
ERROR: Machine: ip-172-31-30-46.us-west-2.compute.internal does not exist!
Some machines have not yet completed launching in the migration account
True

Comment (twarnock2w) [reply cloudendure/cloudendure.py]: Checked code back in.

File: cloudendure/cloudendure.py (hunk 4 above)
Comment (twarnock2w) [reply cloudendure/cloudendure.py]: It's a loop, so just need to store the overall success/failure in a variable to return.

File: cloudendure/cloudendure.py (hunk 2 above)
Comment (twarnock2w) [reply cloudendure/cloudendure.py]: I had it like that so will put it back.

**Comment by mbeacom** [comment]
lgtm

**Pull request merged**